src/app/modules/submission/encrypt-submission/encrypt-submission.controller.ts

```diff
@@ -976,3 +986,113 @@ export const handleGetMetadata = [
   }),
   getMetadata,
 ] as ControllerHandler[]
+
+/**
+ * Handler for POST /:formId/submissions/storage/get-s3-presigned-post-data
+ * Used by handleGetS3PresignedPostData after joi validation
+ * @returns 200 with array of presigned post data
+ * @returns 400 if ids are invalid or total file size exceeds 20MB
+ * @returns 500 if presigned post data cannot be retrieved or any other errors occur
+ */
+const getS3PresignedPostData: ControllerHandler<
+  unknown,
+  AttachmentPresignedPostDataMapType[] | ErrorDto,
+  AttachmentSizeMapType[]
+> = async (req, res) => {
+  const logMeta = {
+    action: 'handleGetS3PresignedPostData',
+    ...createReqMeta(req),
+  }
+
+  return getFeatureFlag(featureFlags.encryptionBoundaryShiftVirusScanner)
+    .map((virusScannerEnabled) => {
+      if (!virusScannerEnabled) {
+        logger.warn({
+          message: 'Virus scanning has not been enabled.',
+          meta: logMeta,
+        })
+
+        const { statusCode, errorMessage } = mapRouteError(
+          new FeatureDisabledError(),
+        )
+        return res.status(statusCode).json({
+          message: errorMessage,
+        })
+      }
+    })
+    .mapErr((error) => {
+      logger.error({
+        message: 'Error retrieving feature flags.',
+        meta: logMeta,
+        error,
+      })
+      const { statusCode, errorMessage } = mapRouteError(error)
+      return res.status(statusCode).json({
+        message: errorMessage,
+      })
+    })
+    .andThen(() =>
+      getQuarantinePresignedPostData(req.body)
+        .map((presignedUrls) => {
+          logger.info({
+            message: 'Successfully retrieved quarantine presigned post data.',
+            meta: logMeta,
+          })
+          return res.json(presignedUrls)
+        })
+        .mapErr((error) => {
+          logger.error({
+            message: 'Failure getting quarantine presigned post data.',
+            meta: logMeta,
+            error,
+          })
+          const { statusCode, errorMessage } = mapRouteError(error)
+          return res.status(statusCode).json({
+            message: errorMessage,
+          })
+        }),
+    )
+}
+
+/**
+ * Custom validation function for Joi to validate that the sum of 'size' in the array of objects
+ * is less than or equal to total file size limit (20MB).
+ */
+const validateFileSizeSum = (
+  value: { size: number }[],
+  helpers: { error: (arg0: string) => null },
+) => {
+  const sum = value.reduce((acc, curr) => acc + curr.size, 0)
+
+  if (sum <= fileSizeLimitBytes(FormResponseMode.Encrypt)) {
+    return value // Return the validated value if the sum of 'size' is less than or equal to 10
```

Comment:
/nit: should be limit, not `10`

Reply:
Oop sorry, fixed in [5e64e4a](https://github.com/opengovsg/FormSG/pull/6685/commits/5e64e4a5fa265200f0fd1862855b235beb7ba2a6), thanks!